models/User.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+//const jwt = require('jsonwebtoken');
+const UserSchema = new mongoose.Schema({
+    name:{
+        type: String,
+        required: [true, 'Please provide your name'],
+        minlength: 3,
+        maxlength: 50,
+    },
+    email:{
+        type: String,
+        required: [true, 'Please provide your email'],
+        match: [
+            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/, 
+            'Please, provide valid email',
+        ],
+        unique: true,
+    },
+    password:{
+        type: String,
+        required: [true, 'Please provide your password'],
+        minlength: 6,
+    },
+});
+
+UserSchema.pre('save', async function(){
+    const salt = await bcrypt.genSalt(10);
+    this.password = await bcrypt.hash(this.password, salt);
+});
+
+UserSchema.methods.createJWT = function(){
```

Comment:
Remember we’re replacing the JWT auth with passport so you’ll want to remove this method and jwt references